### PR TITLE
AArch64: Implement constructor of ARM64PrivateLinkage

### PR DESCRIPTION
**Pull request by knn-k**
This commit implements the constructor of ARM64PrivateLinkage.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
This depends on https://github.com/eclipse/omr/pull/3787.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 1]
```diff
@@ -23,13 +23,94 @@
...
+   _properties._registerFlags[TR::RealRegister::x16]   = ARM64_Reserved; // IP0
+   _properties._registerFlags[TR::RealRegister::x17]   = ARM64_Reserved; // IP1
+
+   for (i = TR::RealRegister::x18; i <= TR::RealRegister::x28; i++)
+      _properties._registerFlags[i] = Preserved; // x18 - x28 Preserved
+
+   _properties._registerFlags[TR::RealRegister::x29]   = ARM64_Reserved; // FP
+   _properties._registerFlags[TR::RealRegister::x30]   = ARM64_Reserved; // LR
+   _properties._registerFlags[TR::RealRegister::sp]    = ARM64_Reserved;
+   _properties._registerFlags[TR::RealRegister::xzr]   = ARM64_Reserved;
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: * x19 (vmthread) and x20 (Java stack pointer) should also be reserved

* in fact, both registers should be "locked" (sorry, I missed that before I merged #5567).  Would you mind fixing that.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 2]
```diff
@@ -23,13 +23,94 @@
...
+   _properties._argumentRegisters[14] = TR::RealRegister::v6;
+   _properties._argumentRegisters[15] = TR::RealRegister::v7;
+
+   _properties._firstIntegerReturnRegister = 0;
+   _properties._firstFloatReturnRegister   = 1;
+
+   _properties._returnRegisters[0]  = TR::RealRegister::x0;
+   _properties._returnRegisters[1]  = TR::RealRegister::v0;
+
+   _properties._numAllocatableIntegerRegisters = 27;
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Should be `25` if you exclude the vmthread and the Java stack pointer.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 3]
```diff
@@ -23,13 +23,94 @@
...
+   _properties._numAllocatableFloatRegisters   = 32;
+
+   _properties._preservedRegisterMapForGC   = 0x00000000; // To be determined
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: I'm looking into this.  I'll get back to you soon.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 1 above)
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Reserved and locked `x19` and `x20`.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 2 above)
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Changed the number.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 4]
```diff
@@ -23,13 +23,99 @@
...
+   for (i = TR::RealRegister::x8; i <= TR::RealRegister::x15; i++)
+      _properties._registerFlags[i] = 0; // x8 - x15 volatile
+
+   _properties._registerFlags[TR::RealRegister::x16]   = ARM64_Reserved; // IP0
+   _properties._registerFlags[TR::RealRegister::x17]   = ARM64_Reserved; // IP1
+
+   _properties._registerFlags[TR::RealRegister::x18]   = Preserved;
+
+   _properties._registerFlags[TR::RealRegister::x19]   = ARM64_Reserved; // vmThread
+   _properties._registerFlags[TR::RealRegister::x30]   = ARM64_Reserved; // Java SP
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: This should be `::x20`.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 3 above)
Comment (0xdaryl) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: I believe this should be 0x1fe40000

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 4 above)
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Bonehead.  Corrected.

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp (hunk 3 above)
Comment (knn-k) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp]: Thank you.  Reflected the value in the code.

**Comment by 0xdaryl** [comment]
Jenkins tests not required.  Merging.

**Pull request merged**